### PR TITLE
Added bundle lifecycle in examples, added link from concepts page

### LINKED ISSUE #627: Showcase the lifecycle of a bundle (via GitRepo CR) from creation to deployment (and updates upon new commits)
We should answer:
- how does a connection happen from token creation to downstream agent bundle deployment?
- what happens when a new commit is pushed? how does Fleet see that change and how does that affect the bundle?
- what are the pods that the bundle information is passed through to the eventual target?

This can likely be easier to manage if we base this off of the [multi-cluster/helm](https://github.com/rancher/fleet-examples/tree/master/multi-cluster/helm) example and use it as the example payload.

### PR DESCRIPTION
**Pull request by jtravee**
Closes #627

References: 
https://fleet.rancher.io/examples/ - added lifecycle under examples page
https://fleet.rancher.io/concepts/ - added link to new section

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+
+1. User creates a [GitRepo](./gitrepo-add.md#create-gitrepo-instance) that points to the [multi-cluster/helm](https://github.com/rancher/fleet-examples/tree/master/multi-cluster/helm) repository.
+2. The Gitjob controller will sync changes from the GitRepo and detect changes from the poll/webhook. With every commit change, the Gitjob controller will create a job that clones the git repository, reads content from the repo such as fleet.yaml and other manifests, and creates the Fleet [bundle](./cluster-bundles-state.md#bundles).
+3. The Fleet-controller then syncs changes from the bundle. According to the targets, the Fleet-controller will create `BundleDeployment` resources, which are a combination of a bundle and a target cluster.
+4. The Fleet-agent will then pull the `BundleDeployment` from the Fleet controlplane. The agent deploys a real application and configuration as a [Helm chart](https://helm.sh/docs/intro/install/) from the `BundleDeployment` into the downstream clusters.
+5. The Fleet-agent will continue to monitor the application bundle and report statuses back in the following order: bundledeployment > bundle > GitRepo > cluster.
```
Comment (nickgerace) [inline docs/examples.md:11]: Question to both @jtravee @StrongMonkey: I think we should standardize on one of the following for Fleet components:

- Fleet-controller
- `fleet-controller`
- fleet-controller
- fleet controller
- Fleet controller
- `fleet` Controller
- `fleet` controller

Personally speaking, I like calling things as a user would see them in a cluster. In that case, that would be `fleet-controller`. However, I could also see a reason to call it the Fleet controller or the controller in Fleet based on the context. Thoughts?

File: docs/examples.md [hunk 2]
```diff
@@ -1,4 +1,16 @@
 # Examples
 
-Examples using raw Kubernetes YAML, Helm charts, Kustomize and combinations
+### Lifecycle of a Fleet Bundle
+
+To demonstrate the lifecycle of a Fleet bundle, we will use multi-cluster/helm as a case study.
+
+1. User creates a [GitRepo](./gitrepo-add.md#create-gitrepo-instance) that points to the [multi-cluster/helm](https://github.com/rancher/fleet-examples/tree/master/multi-cluster/helm) repository.
+2. The Gitjob controller will sync changes from the GitRepo and detect changes from the poll/webhook. With every commit change, the Gitjob controller will create a job that clones the git repository, reads content from the repo such as fleet.yaml and other manifests, and creates the Fleet [bundle](./cluster-bundles-state.md#bundles).
```
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply docs/examples.md:8]: Yes, but the name is generated. Maybe we can call the job with image name so it is more specific.

File: docs/examples.md (hunk 1 above)
Comment (StrongMonkey) [reply docs/examples.md:11]: :exploding_head:

File: docs/examples.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (StrongMonkey) [reply docs/examples.md:8]: I am thinking of this... @jtravee Maybe we can add something like this in the end
> With every commit change, the Gitjob controller will create a job that clones the git repository, reads content from the repo such as fleet.yaml and other manifests, and creates the Fleet [bundle](./cluster-bundles-state.md#bundles). You can found the job pod under the same namespace of gitRepo with image name `rancher/tekton-utils`.

File: docs/examples.md (hunk 1 above)
Comment (StrongMonkey) [reply docs/examples.md:11]: I will leave that to @jtravee since she is the expert here.

Comment (jtravee) [reply docs/examples.md:11]: I agree with `fleet-controller` as the standard!

File: docs/examples.md [hunk 3]
```diff
@@ -1,4 +1,16 @@
 # Examples
 
-Examples using raw Kubernetes YAML, Helm charts, Kustomize and combinations
+### Lifecycle of a Fleet Bundle
+
+To demonstrate the lifecycle of a Fleet bundle, we will use multi-cluster/helm as a case study.
+
+1. User creates a [GitRepo](./gitrepo-add.md#create-gitrepo-instance) that points to the [multi-cluster/helm](https://github.com/rancher/fleet-examples/tree/master/multi-cluster/helm) repository.
```
[the remark this answers is not in the document]
Comment (jtravee) [reply docs/examples.md]: Fixed per feedback.

File: docs/examples.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jtravee) [reply docs/examples.md]: Added.

File: docs/examples.md [hunk 4]
```diff
@@ -1,4 +1,19 @@
...
+
+1. User will create a [GitRepo](./gitrepo-add.md#create-gitrepo-instance) that points to the multi-cluster/helm repository.
+2. The `gitjob-controller` will sync changes from the GitRepo and detect changes from the poll/webhook. With every commit change, the `gitjob-controller` will create a job that clones the git repository, reads content from the repo such as `fleet.yaml` and other manifests, and creates the Fleet [bundle](./cluster-bundles-state.md#bundles).
+
+>**Note:** The job pod with the image name `rancher/tekton-utils` will be under the same namespace as the GitRepo.
+
+3. The `fleet-controller` then syncs changes from the bundle. According to the targets, the `fleet-controller` will create `BundleDeployment` resources, which are a combination of a bundle and a target cluster.
+4. The `fleet-agent` will then pull the `BundleDeployment` from the Fleet controlplane. The agent deploys a real application and configuration as a [Helm chart](https://helm.sh/docs/intro/install/) from the `BundleDeployment` into the downstream clusters.
```
Comment (StrongMonkey) [inline docs/examples.md:13]: > The agent deploys a real application and configuration
Now I feel like I feel bad about words...

> The agent deploys bundle manifests as a helm chart

Comment (jtravee) [reply docs/examples.md]: Sounds good, revised!

File: docs/examples.md [hunk 5]
```diff
@@ -1,4 +1,19 @@
 # Examples
 
-Examples using raw Kubernetes YAML, Helm charts, Kustomize and combinations
+### Lifecycle of a Fleet Bundle
+
+To demonstrate the lifecycle of a Fleet bundle, we will use [multi-cluster/helm](https://github.com/rancher/fleet-examples/tree/master/multi-cluster/helm) as a case study.
+
+1. User will create a [GitRepo](./gitrepo-add.md#create-gitrepo-instance) that points to the multi-cluster/helm repository.
+2. The `gitjob-controller` will sync changes from the GitRepo and detect changes from the poll/webhook. With every commit change, the `gitjob-controller` will create a job that clones the git repository, reads content from the repo such as `fleet.yaml` and other manifests, and creates the Fleet [bundle](./cluster-bundles-state.md#bundles).
```
Comment (StrongMonkey) [inline docs/examples.md:8]: > from the poll/webhook

by polling or webhook event.

@jtravee I think this is better, what do you think? And also can we add this link https://fleet.rancher.io/webhook/ to webhook event?

Comment (jtravee) [reply docs/examples.md]: Revised per feedback.

**Review by StrongMonkey — APPROVED** [review]

**Pull request merged**